message/models.py

```diff
@@ -0,0 +1,57 @@
+from django.db import models
+from django.utils import timezone
+from dogowner.models import DogOwner
+from daycare.models import DayCare
+
+
+class Message(models.Model):
+
+    author = models.CharField(max_length=200)
+    dogowner_id = models.ForeignKey(DogOwner, on_delete=models.DO_NOTHING)
+    daycare_id = models.ForeignKey(DayCare, on_delete=models.CASCADE)
+    text = models.TextField()
+    date = models.DateTimeField(default=timezone.now)
+    Message_Reply = models.ForeignKey('self', on_delete=models.CASCADE, null=True)
+
+    def __str__(self):
+        res = f"Author: {self.author}"
+        res += f" Dog owner ID: {self.dogowner_id}"
+        res += f" Daycare ID: {self.daycare_id}"
+        res += f" text: {self.text}"
+        return res
+
+    @staticmethod
+    def create(author, dogowner_id, daycare_id, text):
+        last = Message.get_last_message_between_dogowner_daycare(
+            dogowner_id, daycare_id)
+
+        if(last is not None):
+            last = last.last()
+
+        new_message = Message(author=author,
+                              dogowner_id=dogowner_id,
+                              daycare_id=daycare_id,
+                              text=text,
+                              Message_Reply=last)
+
+        new_message.save()
+        return new_message
+
+    @staticmethod
+    def get_all_dog_owner_message(user_id):
+        filter_args = {
+            'dogowner_id': user_id,
+        }
+
+        return Message.objects.filter(**filter_args)
+
+    @staticmethod
+    def get_all_daycare_message(user_id):
+        filter_args = {
+            'daycare_id': user_id,
+        }
+        return Message.objects.filter(**filter_args)
+
+    @staticmethod
+    def get_last_message_between_dogowner_daycare(dogowner_id, daycare_id):
+        return Message.objects.filter(dogowner_id=dogowner_id, daycare_id=daycare_id)
```

Comment:
There is an option that the daycare and the dogowner didn't started any chat, maybe we should return null or something like that in that case

Reply:
If there are no messages  the filter return none anyway

Reply:
You can check the call for this func in message.create function